src/server/qgsserverprojectutils.h

```diff
@@ -306,16 +308,18 @@ namespace QgsServerProjectUtils
   SERVER_EXPORT QStringList wmsRestrictedComposers( const QgsProject &project );
 
   /**
-   * Returns the WMS service url defined in a QGIS project.
+   * Returns the WMS service url.
    * \param project the QGIS project
-   * \returns url if defined in project, an empty string otherwise.
+   * \param request the request
+   * \param settings the server settings
+   * \returns url to use for this service
    */
-  SERVER_EXPORT QString wmsServiceUrl( const QgsProject &project );
+  SERVER_EXPORT QString wmsServiceUrl( const QgsProject &project, const QgsServerRequest &request = QgsServerRequest(), const QgsServerSettings *settings = NULL );
```

Comment:
It should be `nullptr` instead of `NULL` but I'd prefer to use const reference instead of pointer (if possible).

Reply:
(same comments for wfs, wmts, ... methods)

Reply:
Try with const reference